### PR TITLE
Add option `exclude_attrs` to processing.parameter_as_dict() 

### LINKED ISSUE #824: Add ability to exclude attributes from processing results
This issue is related to #707

Problem description:
When using oemof_tabulars facades energysystem components are generated with multiple attributes which are not related to oemof.solph. Using `processing.parameter_as_dict()` of oemof.solph results in many unwanted parameters from related component (i.e. `subnodes` as written in #707). It would be nice to have an option to exclude attributes from parameter dict generation.

Proposal:
Add attribute `exclude_attrs=None` to `parameter_as_dict()` which can hold a list of unwanted attributes, which are then filtered out before preparing parameter dict results. Default results are untouched as `exclude_attrs` is optional.

### PR DESCRIPTION
**Pull request by henhuy**
This PR adds an option to exclude attributes from `processing.parameter_as_dict()` scan in.

Closes #824

### CODE REVIEW

**Comment by henhuy** [comment]
Test fails due to black fails in unrelated files... 
What to do about this?

**Review by jnnr — COMMENTED** [review]
Thanks for this feature! Looks good. Have not tested it. The fails of black in unrelated files doesn't have to be solved here.

File: src/oemof/solph/processing.py [hunk 1]
```diff
@@ -251,7 +253,15 @@ def __separate_attrs(system, get_flows=False, exclude_none=True):
     {(node1, node2): {'scalars': {'attr1': scalar, 'attr2': 'text'},
     'sequences': {'attr1': iterable, 'attr2': iterable}}}
 
-    om : A solved oemof.solph.Model.
+    system:
+        A solved oemof.solph.Model or oemof.solph.Energysystem
+    exclude_attrs: List[str]
+        Optional list of additional attributes which shall be excluded from
+        parameter dict
```
Comment (p-snft) [inline src/oemof/solph/processing.py:260]: The documentation states that the argument is optional, but as no default (`None`?) is given, it is obligatory.

Comment (henhuy) [reply src/oemof/solph/processing.py:260]: You are right. I moved initialization of `exclude_attrs` into function `parameter_as_dict` where it is marked correctly as optional and forgot about init in `separate_attrs`. Will add it there too. Thanks

File: src/oemof/solph/processing.py [hunk 2]
```diff
@@ -369,6 +369,8 @@ def remove_nones(com):
             if len(value) == 0 or value[0] is None:
                 del com["sequences"][ckey]
 
+    exclude_attrs = exclude_attrs or []
```
Comment (p-snft) [inline src/oemof/solph/processing.py:372]: In my opinion, a classical
```suggestion
    if exclude_attrs is None:
        exclude_attrs = []
```
would be more comprehensible.

Comment (henhuy) [reply src/oemof/solph/processing.py]: okay for me - I like this style, but adapted your suggestion (for both places)

**Review by p-snft — APPROVED** [review]
Fine by me.

**Pull request merged**